obspy/signal/trigger.py

```diff
@@ -303,12 +303,16 @@ def energy_ratio(a, nsta):
 
     .. seealso:: [Han2009]_
     """
+    if nsta > len(a) // 2 or nsta == 0:
+        # Half forward, half backward -> empty medium
+        # If nsta is zero, the sum is undefined -> return zero
+        return np.zeros(len(a), dtype=np.float64)
```

Comment:
Shouldn't `nsta=0` be considered nonsensical input and rather raise a `ValueError` maybe? Any point to allowing `nsta` to be zero? @dizcza

Reply:
Actually, I fell the same about `nsta` being longer than half the signal, I think that also should raise a `ValueError`, which would make it much easier for people to wrap a `try/except ValueError` around their code in case of weird signal input (like gappy stream with very short data parts), although I guess this could be treated on a higher level, checking e.g. traces in a stream before running the trigger on them.

Reply:
> Any point to allowing nsta to be zero?

No.

Perhaps, indeed raising an error would be better.

Reply:
Some of these routines are from the very early days, so there can be code that is close to 14 years old, so not everything what's in there is 100% up to par. Where looking at new code much more closely nowadays, for the most part. :smile: